api_tests/base/test_sloan_study.py

```diff
@@ -0,0 +1,210 @@
+        assert SLOAN_COI in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=True; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=True; Path=/' in cookies
+        assert f' {SLOAN_COI}=True; Path=/' in cookies
+
+    @pytest.mark.enable_quickfiles_creation
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_sloan_study_variable_unauth(self, mock_flag_is_active, app, user, preprint):
+        mock_flag_is_active.return_value = True
+        headers = {'Referer': preprint.absolute_url}
+        resp = app.get('/v2/', headers=headers)
+
+        assert SLOAN_DATA in resp.json['meta']['active_flags']
+        assert SLOAN_PREREG in resp.json['meta']['active_flags']
+        assert SLOAN_COI in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=True; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=True; Path=/' in cookies
+        assert f' {SLOAN_COI}=True; Path=/' in cookies
+
+    @pytest.mark.enable_quickfiles_creation
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_sloan_study_control(self, mock_flag_is_active, app, user, preprint):
+        mock_flag_is_active.return_value = False
+        headers = {'Referer': preprint.absolute_url}
+        resp = app.get('/v2/', auth=user.auth, headers=headers)
+
+        tags = user.all_tags.all().values_list('name', flat=True)
+
+        assert f'no_{SLOAN_DATA}' in tags
+        assert f'no_{SLOAN_PREREG}' in tags
+        assert f'no_{SLOAN_COI}' in tags
+
+        assert SLOAN_DATA not in resp.json['meta']['active_flags']
+        assert SLOAN_PREREG not in resp.json['meta']['active_flags']
+        assert SLOAN_COI not in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=False; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=False; Path=/' in cookies
+        assert f' {SLOAN_COI}=False; Path=/' in cookies
+
+    @pytest.mark.enable_quickfiles_creation
+    @mock.patch('api.base.views.Flag.is_active')
+    def test_sloan_study_control_unauth(self, mock_flag_is_active, app, user, preprint):
+        mock_flag_is_active.return_value = False
+        headers = {'Referer': preprint.absolute_url}
+        resp = app.get('/v2/', headers=headers)
+
+        assert SLOAN_DATA not in resp.json['meta']['active_flags']
+        assert SLOAN_PREREG not in resp.json['meta']['active_flags']
+        assert SLOAN_COI not in resp.json['meta']['active_flags']
+
+        cookies = resp.headers.getall('Set-Cookie')
+
+        assert f' {SLOAN_DATA}=False; Path=/' in cookies
+        assert f' {SLOAN_PREREG}=False; Path=/' in cookies
+        assert f' {SLOAN_COI}=False; Path=/' in cookies
+
+    @pytest.mark.parametrize('reffer_url', [
+        f'{DOMAIN}preprints',
+        f'{DOMAIN}preprints/',
+        f'{DOMAIN}preprints/foorxiv',
```

Comment:
because of the regex bug above, i'd guess this referer will actually map to the `osf` provider, and this test merely asserts that it matched *any* provider.

if you give `foorxiv` and `osf` different sloan preferences, you could tell the difference and assert that the cookies match the expected provider.